sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -654,7 +659,307 @@ void setAndGetSessionState(MessagingEntityType entityType) {
     }
 
     /**
-     * Sets the sender and receiver. If session is enabled, then a single-named session receiver is created.
+     * Verifies that we can not receive from a session enabled receiver which try to connect to non session entity.
+     */
+    @MethodSource("messagingEntityProvider")
+    @ParameterizedTest
+    void createSessionReceiverForNonSessionEntity(MessagingEntityType entityType) {
+        // Arrange
+        String sessionId = "test-session-id";
+        ServiceBusClientBuilder.ServiceBusSessionReceiverClientBuilder builder = getBuilder(false)
+            .sessionReceiver();
+
+        if (entityType == MessagingEntityType.QUEUE) {
+            builder.queueName(getQueueName());
+        } else if (entityType == MessagingEntityType.SUBSCRIPTION) {
+            builder.topicName(getTopicName());
+            builder.subscriptionName(getSubscriptionName());
+        }
+
+        ServiceBusReceiverAsyncClient invalidReceiver = builder.sessionId(sessionId).buildAsyncClient();
+
+        // Assert & Act
+        StepVerifier.create(invalidReceiver.receive())
+            .verifyErrorMatches(error -> error instanceof UnsupportedOperationException);
+    }
+
+
+    /**
+     * Send multiple message and receive from multiple receivers and autocomplete is off.
+     */
+    @Disabled("Last message does not arrive : https://github.com/Azure/azure-sdk-for-java/issues/10840")
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void multipleReceiverAndClientComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+
+        int firstBatch = 2;
+        int secondBatch = 2;
+        final String messageId = UUID.randomUUID().toString();
+        final List<ServiceBusMessage> messages;
+        if (isSessionEnabled) {
+            messages = TestUtils.getServiceBusMessages(firstBatch, CONTENTS_BYTES, messageId, isSessionEnabled,
```

Comment:
I assume that if the method overload with sessionId is passed, then we want to add it to the message. parameter `isSessionEnabled` isn't necessary.